### PR TITLE
[mlir][sparse] add ReinterpretMapScopeOption for the pass

### PR DESCRIPTION
**Pull request by PeimingLiu**

### CODE REVIEW

File: mlir/include/mlir/Dialect/SparseTensor/Transforms/.#Passes.h [hunk 1]
```diff
@@ -0,0 +1 @@
+peiming@peiming.c.googlers.com.13658:1697570271
```
Comment (yinying-lisa-li) [inline mlir/include/mlir/Dialect/SparseTensor/Transforms/.#Passes.h:1]: Is this needed?

File: mlir/include/mlir/Dialect/SparseTensor/Transforms/Passes.h [hunk 2]
```diff
@@ -40,6 +40,14 @@ enum class SparseParallelizationStrategy {
   kAnyStorageAnyLoop
 };
 
+/// Define a scope for reinterpret map pass.
```
Comment (yinying-lisa-li) [inline mlir/include/mlir/Dialect/SparseTensor/Transforms/Passes.h:43]: nit: defines

File: mlir/include/mlir/Dialect/SparseTensor/Transforms/Passes.h [hunk 3]
```diff
@@ -40,6 +40,14 @@ enum class SparseParallelizationStrategy {
...
+enum class ReinterpretMapScope {
+  kAll,           // reinterpret all applicable operations.
```
Comment (yinying-lisa-li) [inline mlir/include/mlir/Dialect/SparseTensor/Transforms/Passes.h:45]: nit: reinterprets

File: mlir/include/mlir/Dialect/SparseTensor/Transforms/.#Passes.h (hunk 1 above)
Comment (PeimingLiu) [reply mlir/include/mlir/Dialect/SparseTensor/Transforms/.#Passes.h:1]: NO! Emacs generated file, will delete it.

File: mlir/include/mlir/Dialect/SparseTensor/Transforms/Passes.h [hunk 4]
```diff
@@ -40,6 +40,14 @@ enum class SparseParallelizationStrategy {
   kAnyStorageAnyLoop
 };
 
+/// Defines a scope for reinterpret map pass.
+enum class ReinterpretMapScope {
+  kAll,           // reinterprets all applicable operations
+  kGenericOnly,   // reinterprets only linalg.generic
+  kExceptGeneric, // reinterprets operation other than linalg.generic (e.g.,
```
Comment (aartbik) [inline mlir/include/mlir/Dialect/SparseTensor/Transforms/Passes.h:47]: can you try to keep this on one line? I like // xxx comments to stay unbroken (if possible)

**Review by aartbik — APPROVED** [review]

**Pull request merged**